pkgs/games/am2rlauncher/default.nix

```diff
@@ -0,0 +1,113 @@
+{ lib,
+  buildDotnetModule,
+  pkgsi686Linux,
+  glibc,
+  gtk3,
+  libappindicator,
+  webkitgtk,
+  e2fsprogs,
+  libnotify,
+  libgit2,
+  openssl,
+  xdelta,
+  file,
+  busybox,
+  openjdk,
+  steam-run,
+  patchelf,
+  fetchFromGitHub,
+  makeDesktopItem }:
+
+buildDotnetModule rec {
+  pname = "AM2RLauncher";
+  version = "2.3.0";
+  src = fetchFromGitHub {
+    owner = "AM2R-Community-Developers";
+    repo = pname;
+    rev = "c04bad8a331a4126d93e17020d090a2c7ee63c72";
+    sha256 = "sha256-/nHqo8jh3sOUngbpqdfiQjUWO/8Uzpc5jtW7Ep4q6Wg=";
+  };
+
+  projectFile = [
+    "AM2RLauncher/AM2RLauncher.Gtk/AM2RLauncher.Gtk.csproj"
+  ];
+
+  nugetDeps = ./deps.nix;
+  executables = "AM2RLauncher.Gtk";
+
+  runtimeDeps = [
+    # needed for launcher
+    glibc
+    gtk3
+    libappindicator
+    webkitgtk
+    e2fsprogs
+    libnotify
+    libgit2
+    openssl
+
+    # needed for 32-bit game binary
+    pkgsi686Linux.glibc
+    pkgsi686Linux.stdenv.cc.cc.lib
+    pkgsi686Linux.zlib
+    pkgsi686Linux.xorg.libXxf86vm
+    pkgsi686Linux.libGL
+    pkgsi686Linux.openal
+    pkgsi686Linux.libpulseaudio
+    pkgsi686Linux.xorg.libXext
+    pkgsi686Linux.xorg.libX11
+    pkgsi686Linux.xorg.libXrandr
+    pkgsi686Linux.libGLU
+  ];
+
+  buildInputs = [
+    gtk3
+  ];
+
+  patches = [
+    ./steam-run-binary.patch
+  ];
+
+  dotnetFlags = [
+    "-p:DefineConstants=\"NOAPPIMAGE;NOAUTOUPDATE;PATCHOPENSSL\""
+  ];
+
+  postFixup = ''
+    wrapProgram $out/bin/AM2RLauncher.Gtk \
+      --prefix PATH : ${lib.makeBinPath [
+        xdelta
+        file
+        busybox
+        openjdk
+        steam-run
+        patchelf
+      ]} \
+
+    mkdir -p $out/share/icons
+    install -Dm644 $src/AM2RLauncher/distribution/linux/AM2RLauncher.png $out/share/icons/am2r.png
+  '';
+
+  desktopItems = [(makeDesktopItem {
+    desktopName = "AM2R Launcher";
+    name = "am2rlauncher";
+    exec = "AM2Rlauncher.Gtk";
+    icon = "am2r";
+    comment = meta.description;
+    type = "Application";
+    categories = [ "Game" ];
+  })];
```

Comment:
There is already a .desktop file in https://github.com/AM2R-Community-Developers/AM2RLauncher/blob/main/AM2RLauncher/distribution/linux/AM2RLauncher.desktop

The exec there is `AM2RLauncher` instead of `AM2RLauncher.Gtk`, though. The [AUR package](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=am2rlauncher) (linked in the README, I don't know if it's official) renames the executable to fit the .desktop file.